lib/ml/feature_label.hpp

```diff
@@ -14,55 +14,29 @@
 
 #pragma once
 
-#include <cassert>
-#include <cmath>
-#include <string>
-#include <utility>
-#include <vector>
-
-#include "boost/tokenizer.hpp"
-#include "core/engine.hpp"
+#include "lib/vector.hpp"
 
 namespace husky {
 namespace lib {
 namespace ml {
-using vec_double = std::vector<double>;
 
-template <typename FT, typename LT>
-class FeatureLabelBase {
+// extends LabeledPoint, using Vector to represent features
+template <typename FeatureT, typename LabelT, bool is_sparse>
+class FeatureLabel : public LabeledPoint<Vector<FeatureT, is_sparse>, LabelT> {
```

Comment:
I think it is not a good idea to leave it TODO because many things will depend on it soon. It would be blocking the development of other machine leaning algorithm.

BTW, it should be a general extension of LabeledPoint, so may be we can put them together.